Assets/Scripts/EncounterScripts/TileEncounterTrigger.cs

```diff
@@ -6,10 +6,12 @@ public class TileEncounterTrigger : MonoBehaviour
 {
 	[SerializeField]
 	private Encounter _encounter;
+	[SerializeField]
 	private EncounterManager _encounterManager;
 
 	private void OnTriggerEnter2D(Collider2D collision)
 	{
```

Comment:
this should check that the player is the one triggering it, not just any collision